### PR TITLE
calculate & display cost

### PR DESCRIPTION
**Pull request by kshepetska**
Overwiev:

https://github.com/fs-jan24-codeup/phone-catalog/assets/155641185/54704856-0934-49b8-8131-62692f8b2a71

### CODE REVIEW

**Review by Vasya564 — CHANGES_REQUESTED** [review]
Looks good! But require some improvements

File: src/components/CartItem/CartItem.scss [hunk 1]
```diff
@@ -103,8 +108,23 @@
...
+
+    &::before {
+        content: '$';
+        position: absolute;
+        left: 45px;
+    }
+}
+
+
+
```
Comment (Vasya564) [inline src/components/CartItem/CartItem.scss:125]:
```suggestion
    &::before {
        content: '$';
    }
}
```

File: src/components/CartItem/CartItem.tsx [hunk 2]
```diff
@@ -7,40 +7,52 @@ import Photo from '../../assets/icons/Photo.svg';
...
-}
+        <div className='cart-item__img'>
+          <img src={Photo} alt="Photo"/>
+        </div>
+        <p className='cart-item__name'>{name}</p>
+        <div className='cart-item__buttons'>
+          <div className='cart-item__buttons-icons'>
+            <button className='cart-item__buttons-icon' onClick={handleClickMinus}>
+               <img src={Minus} alt="minus" />
+            </button>
```
Comment (Vasya564) [inline src/components/CartItem/CartItem.tsx:45]: Button should be disabled if ``quantity === 1``

File: src/components/CartItem/CartItem.scss [hunk 3]
```diff
@@ -80,11 +80,16 @@
...
 
 .cart-item__buttons-icon {
-    width: 16px;
-    height: 16px;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    border: none;
+    background: none;
+    outline: none;
```
Comment (Vasya564) [inline src/components/CartItem/CartItem.scss:92]: To center icon inside button simply add
```
display: flex;
align-items: center;
justify-content: center; 
```
to .cart-item__buttons-icons class

**Review by Vasya564 — CHANGES_REQUESTED** [review]

File: src/components/CartItem/CartItem.scss (hunk 3 above)
Comment (kshepetska) [reply src/components/CartItem/CartItem.scss:92]: Monosnap Vite + React + TS - Google Chrome 2024-04
It have default styles

File: src/components/CartItem/CartItem.tsx [hunk 4]
```diff
@@ -44,24 +47,20 @@ export const CartItem: React.FC<Props> = ({ item }) => {
...
-            <img
-              src={Minus}
-              alt="minus"
-              className="cart-item__buttons-icon"
-              onClick={handleClickMinus}
-            />
+          {quantity >= 1 && (
+            <button className='cart-item__buttons-icon' onClick={handleClickMinus}>
+               <img src={Minus} alt="minus" />
+            </button>
```
Comment (Vasya564) [inline src/components/CartItem/CartItem.tsx:53]:
```suggestion
            <button 
              className='cart-item__buttons-icon' 
               onClick={handleClickMinus} 
               disabled={quantity === 1}>
               <img src={Minus} alt="minus" />
            </button>
```

File: src/pages/CartPage/CartPage.tsx [hunk 5]
```diff
@@ -4,9 +4,29 @@ import './CartPage.scss';
...
+
+  const totalPrice = cartItems.reduce((total, item) => total + item.price * item.quantity, 0);
+
+  const itemCount = cartItems.reduce((total, item) => total + item.quantity, 0);
+
+  const updateQuantity = (index: number, newQuantity: number) => {
+    const newCartItems = [...cartItems];
+    newCartItems[index].quantity = newQuantity;
+    setCartItems(newCartItems);
+  };
```
Comment (Vasya564) [inline src/pages/CartPage/CartPage.tsx:29]: here also update quantity should use ``id`` instead of ``index``

**Review by Vasya564 — CHANGES_REQUESTED** [review]

**Review by Vasya564 — APPROVED** [review]

**Pull request merged**